apps/contributor/test/contributor.js

```diff
@@ -0,0 +1,160 @@
+              'sender does not have permission'
+            });
+        });
+
+        it('add contributor', async() => {
+            let contributorCount = await contributor.coreContributorsCount();
+            await contributor.addContributor(account, hashDigest, hashFunction, hashSize);
+            assert.equal(await contributor.addressExists(account), true);
+            let contributorCountAfter = await contributor.coreContributorsCount();
+            assert.equal(await contributorCountAfter.toNumber(), parseInt(contributorCount)+1);
+        });
+
+        it("should revert when add contributor with an address that already exist", async() => {
+            return assertRevert(async() => {
+              await contributor.addContributor(account, hashDigest, hashFunction, hashSize, { from: member1})
+              'address already exist'
+            });
+        });
+    });
+
+    describe("Update contributor", async() => {
+        let id;
+        let oldAccount;
+        let newAccount;
+        let hashDigest;
+        let hashFunction;
+        let hashSize;
+
+        before(async() =>  {
+            id = await contributor.coreContributorsCount();
+            oldAccount = root;
+            newAccount = member1;
+            hashDigest = '0x1';
+            hashFunction = 1;
+            hashSize = 1;
+        });
+
+        it('update contributor account', async() => {
+            await contributor.updateContributorAccount(id.toNumber(), oldAccount, newAccount);
+            let contributorId = await contributor.getContributorIdByAddress(oldAccount);
+            assert.equal(contributorId.toNumber(), 0);
+        });
+
+        it("should revert when update contributor account from address that does not have permission", async() => {
+            return assertRevert(async() => {
+                await contributor.updateContributorAccount(id.toNumber(), oldAccount, newAccount, {from: member1});
+              'sender does not have permission'
+            });
+        });
+
+        it("should revert when update contributor account that does not exist", async() => {
+            return assertRevert(async() => {
+              await contributor.updateContributorAccount(id.toNumber(), accounts[3], newAccount);
+              'contributor does not exist'
+            });
+        });
+
+        it("should revert when update contributor account with address(0)", async() => {
+            return assertRevert(async() => {
+              await contributor.updateContributorAccount(id.toNumber(), oldAccount, ZERO_ADDR);
+              'contributor does not exist'
+            });
+        });
+
+        it('update contributor profile hash', async() => {
+            await contributor.updateContributorProfileHash(id.toNumber(), hashDigest, hashFunction, hashSize);
+            let contributorId = await contributor.getContributorIdByAddress(oldAccount);
+            assert.equal(contributorId.toNumber(), 0);
```

Comment:
This doesn't actually check for an updated hash afaics.

Reply:
yup...solved in this [commit](https://github.com/67P/kredits-contracts/pull/106/commits/ab6d6a40251a9589d99886b0e53bb86b74f50080)